src/main/java/org/apache/aurora/scheduler/mesos/DockerContainerTasks.java

```diff
@@ -73,12 +74,33 @@ public static void configureTask(
 		taskBuilder.setCommand(cmd.build());
 
 		// set kill policy
-		if (taskConfig.isSetKillPolicy()) {
-			long killGracePeriodNanos = Duration.ofSeconds(taskConfig.getKillPolicy().getGracePeriodSecs()).toNanos();
+		if (task.getTask().isSetKillPolicy()) {
+			long killGracePeriodNanos = Duration.ofSeconds(task.getTask().getKillPolicy().getGracePeriodSecs()).toNanos();
 			KillPolicy.Builder killPolicyBuilder = KillPolicy.newBuilder()
 					.setGracePeriod(DurationInfo.newBuilder().setNanoseconds(killGracePeriodNanos).build());
 			taskBuilder.setKillPolicy(killPolicyBuilder.build());
 		}
+		
+		// set health check
+		if (task.getTask().isSetHealthCheck()) {
+			IHealthCheck healthCheck = task.getTask().getHealthCheck();
+			HealthCheck.Builder mesosHealthCheck = HealthCheck.newBuilder()
+					.setConsecutiveFailures(healthCheck.getConsecutiveFailures())
+					.setDelaySeconds(healthCheck.getDelaySeconds())
+					.setGracePeriodSeconds(healthCheck.getGracePeriodSeconds())
+					.setIntervalSeconds(healthCheck.getIntervalSeconds())
+					.setTimeoutSeconds(healthCheck.getTimeoutSeconds());
+			if (healthCheck.isSetShell()) {
+				mesosHealthCheck.setCommand(CommandInfo.newBuilder()
+						.setValue(healthCheck.getShell().getCommand()).build());
+			} else if (healthCheck.isSetHttp()) {
```

Comment:
If validation is done before then the "else" is not needed

Reply:
but it makes the intention clear without needing to dig into the validator